src/python/detectors/notebook_variable_redefinition/notebook_variable_redefinition_compliant.ipynb

```diff
@@ -0,0 +1,84 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "####  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\n",
+    "####  SPDX-License-Identifier: Apache-2.0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# {fact rule=notebook-variable-redefinition@v1.0 defects=0}\n",
+    "import math\t"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = int(input('x:'))\n",
+    "y = int(input('y:'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "z = '5'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "z: 44.64302857109943\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Compliant: even though there are 2 different types assigned to variable `z`, it is only used within the same cells.\n",
```

Comment:
Even though there are 2 different types assigned to variable `z` in two different cells, it is only used within one of the two cells.